pkg/controller/operconfig/migration.go

```diff
@@ -97,3 +112,325 @@ func convertEgressFirewallToEgressNetworkPolicy(ctx context.Context, client cnoc
 	}
 	return nil
 }
+
+func convertSdnEgressIpToOvnEgressIp(ctx context.Context, client cnoclient.Client) error {
+
+	// 1. query for hostsubnets and netnamespaces
+	hsnList, err := client.Default().Dynamic().Resource(gvrHostSubnet).List(ctx, metav1.ListOptions{})
```

Comment:
> I think we should write some tests for this.
I agree. This is not a trivial effort however, so I will do this in a separate PR.

Reply:
I do not know how to make `List` calls paginated. Perhaps we can discuss this on Slack privately?

Reply:
> I do not know how to make `List` calls paginated or what that does. Perhaps we can discuss this on Slack privately? For what it's worth, I do not have problems with excessive loads at the moment so I don't know how important this is. EgressFirewall migration code has similar `List` calls and has worked without issue.

See for example
https://github.com/openshift/sdn/blob/master/pkg/network/common/list_pager.go

This is important at scale.

Reply:
> I disagree with this - if there are no tests, how do we know this feature does what we want it to ? And do we ensure it does not break when future PRs are merged ?

I have a few comments in response.
- To be clear, in an ideal world I **100 percent** agree with your reasoning. However, this feature needs to be in by end of this sprint, and one week of development time is lost due to the face-to-face meetup in Raleigh. This makes it even more urgent to get this code in.
- I will not support the merging of this code until the ready-to-merge code has been tested manually by me and nothing is broken on my end.
- To your point about breaking when future code is introduced, I don't think this code (which is only executed during a migration) will even be hit at ALL by the average user. Hypothetically, for any user trying to perform a migration, why are they provisioning an OpenshiftSDN cluster using 4.12 code in the first place? 4.12 has not released yet and so risk is to be expected.
- See review comments on [the egressFirewall PR that has been merged](https://github.com/openshift/cluster-network-operator/pull/1534#issuecomment-1228585457). My PR is significantly larger than the egressFirewall PR and so I believe this code should be held to the same standard, where e2e CI can be provided at a later date (still, I believe CI is *high priority* and is what I shall work on next.
- Unit tests are reasonable to add, but again, requires some effort. I value an e2e test much higher than unit tests so I would rather focus development time on that.

Reply:
> See for example
> https://github.com/openshift/sdn/blob/master/pkg/network/common/list_pager.go
>
> This is important at scale.
I added paginated list calls in my latest commit. I made these changes to ALL list calls in `migration.go` including egressFirewall migration code.

Reply:
> * To be clear, in an ideal world I **100 percent** agree with your reasoning. However, this feature needs to be in by end of this sprint (it is the ONLY release blocker for 4.12), and one week of development time is lost due to the face-to-face meetup in Raleigh. This makes it even more critical to get this code in.

I don't think this is a reason to not do tests. It would be a reason to scope smaller amount of work in a sprint/release. No tests just translates in lower quality code.

> * To your point about breaking when future code is introduced, I don't think this code (_which is only executed during a migration_) will even be hit at ALL by the average user. Hypothetically, for any user trying to perform a migration, why are they provisioning an OpenshiftSDN cluster using 4.12 code in the first place? 4.12 has not released yet and so risk is to be expected.

I don't think this is related to end user usage or circumstances. It's about code maintainability, which needs to happen as long as the code is there.

> * See review comments on [the egressFirewall PR that has been merged](https://github.com/openshift/cluster-network-operator/pull/1534#issuecomment-1228585457). My PR is significantly larger than the egressFirewall PR and so I believe this code should be held to the same standard, where e2e CI can be provided at a later date (still, I believe CI _is_ high priority and it is what I shall work on next.

That was an exception because it was the end of the sprint (while we are not at the end of our sprint) and it was stated tests would be added in a follow-up PR. It might change from being an exception to being a mistake if at the end the tests are never added or if it is used as a precedent to request the same kind of exception multiple other times.

> * Unit tests are reasonable to add, but again, requires some effort. I value an e2e test much higher than unit tests so I would rather focus development time on that. (also e2e tests for this shall live in `openshift/release`)

Writing unit tests is not a separate effort from writing the code itself.

> 
> Ultimately I don't control when code gets merged so if the consensus is that tests are absolutely necessary prior to merge...then hopefully I will be able to provide CI soon.

I am unaware if we have any kind of policy regarding unit tests. I will ask. I just made the suggestion because it seemed to me it would add quality to the PR.

Reply:
I have added unit tests. Please take a look. I would like to aim to have this PR merged by Friday, so any feedback is welcome. @jcaamano @pliurh